src/main/java/net/uaznia/lukanus/hudson/plugins/gitparameter/GitParameterDefinition.java

```diff
@@ -290,10 +300,12 @@ public String prettyRevisionInfo(GitClient newgit, Revision r) {
 					return Collections.singletonMap(errMsg, errMsg);
 				}
 
-				FetchCommand fetch = newgit.fetch_().from(remoteURL,
-						repository.getFetchRefSpecs());
-				fetch.execute();
 				if (type.equalsIgnoreCase(PARAMETER_TYPE_REVISION)) {
+					// If we use the GitClient#getRemoteReferences function, we don't
```

Comment:
I guess I initially wanted to be sure that the new behavior would be an opt-in feature, rather than forced.  Over time, as the feature proves to be stable, it could become the default (or only) behavior.

The "if possible" from the commit message is referring to the fact that you cannot use `ls-remote` for the "Revision" type.  So when the type is set to revision, it would not be possible and must fall back to the workspace behavior.  I believe the approach is functionally identical to yours, save for the opt-in checkbox.